lib/check_params.rb

```diff
@@ -1,19 +1,29 @@
-require "json"
+require 'json'
 
 class CheckParams
-  attr_reader :blog_title, :hash_key, :pages
+  attr_reader :blog_title, :body, :pages
 
-  def initialize(blog_title, hash_key, pages)
+  def initialize(blog_title, body, pages)
     @pages = pages
     @blog_title = blog_title
-    @hash_key = hash_key
+    @body = body
   end
 
   def blog_available?
     pages.keys.include? blog_title
   end
 
-  def key_correct?
-    pages[blog_title]["key"] == hash_key
+  def verify_signature
+    signature = 'sha1=' + OpenSSL::HMAC.hexdigest(
+      OpenSSL::Digest.new('sha1'),
+      pages[blog_title]['key'],
+      body
+    )
+    unless Rack::Utils.secure_compare(
+      signature,
+      request.env['HTTP_X_HUB_SIGNATURE']
```

Comment:
I need help with this line, this was copied over from the github developers page but i dont know where this HTTP_X_HUB_SIGNATURE is comming from